src/index.ts

```
@@ -29,43 +29,47 @@ export class LabelRenderer {
+      }
+    })
+
+    // Update existing or create new labels
+    labels.forEach((label) => {
       const { x, y, fontSize, color, text, weight, opacity, shouldBeShown, style, className } = label
-      const exists = labelsToDelete.get(label.id)
-      if (exists) {
-        labelsToDelete.delete(label.id)
-      } else {
-        const cssLabel = new CssLabel(this._container, label.text)
+      if (isNaN(x) && isNaN(y)) return
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)